### PR TITLE
docs(coverage): guide for README badges

### LINKED ISSUE #2288: Docs on how to create coverage badge
### Clear and concise description of the problem

The [docs on coverage](https://vitest.dev/guide/coverage.html) are a great starting point but don't go into detail.

### Suggested solution

Add a paragraph on how to create a coverage badge.

Example:

![Coverage badge][coverage-badge]

[coverage-badge]: https://img.shields.io/badge/Coverage-100%25-brightgreen.svg

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Resolves #2288

image

### CODE REVIEW

File: docs/guide/coverage.md [hunk 1]
```diff
@@ -125,3 +126,39 @@ if (condition) {
...
+2. Add placeholders into your `README.md` for badges:
+
+```md
+| Statements                  | Branches                | Functions                 | Lines             |
+| --------------------------- | ----------------------- | ------------------------- | ----------------- |
+| ![Statements](#statements#) | ![Branches](#branches#) | ![Functions](#functions#) | ![Lines](#lines#) |
+```
+
+3. Run `istanbul-badges-readme` by adding it as project dependency, or directly with `npx`, e.g. `npx istanbul-badges-readme`.
+4. Placeholders in `README.md` should now be automatically updated with latest coverage results.
```
Comment (dammy001) [inline docs/guide/coverage.md:162]:
```suggestion
4. Placeholders in `README.md` should now be automatically updated with the latest coverage results.
```

**Review by dammy001 — APPROVED** [review]

**Comment by antfu** [comment]
Thanks for your contribution. However, my concern is that, there are many different ways and services that can offer the badges. Documenting one means we are advocating it over the others, which could be unfair.

I would consider the badges a personal preference. I suggest we not mention it to be natural, just like how other testing frameworks are doing.

**Comment by janosh** [comment]
Thanks @AriPerkkio!

**Comment by khalwat** [comment]
For anyone finding this, here's a way to do it with GitHub actions:

Here's a working example with Vitest: https://github.com/nystudio107/rollup-plugin-critical/blob/master/.github/workflows/node.js.yml

This is the npm script it references to build the coverage summary for the above GitHub action:

```json
    "test-ci": "vitest run --coverage.enabled --coverage.reporter='text-summary'",
```

**Pull request closed without merging**